docs/_posts/testing_a_tricky_edge_case_in_a_tuf_client.md

```diff
@@ -0,0 +1,117 @@
+
+We can play with versions and expiry to verify that this scenario not explicitly mentioned in the spec works correctly and safely. By using the simulator, we can do the following:
+1. Set the timestamp expiry one week ahead (Day 7)
+2. On the very first day download, verify, and load metadata for the [top-level roles](https://theupdateframework.github.io/specification/latest/#roles-and-pki) following the TUF specification order. This is done by simply calling `RespositorySimulator.refresh()`.
+3. Then we bump [snapshot](https://theupdateframework.github.io/specification/latest/#update-snapshot) and [targets](https://theupdateframework.github.io/specification/latest/#targets) versions to v2 on the same day
+4. Set v2 expiry dates three weeks ahead (Day 21)
+5. Travel in time somewhere between Day 7 and Day 21
+6. Perform a successful `refresh` (with `RepositorySimulator.refresh()` call) with the expired locally cached timestamp
+7. Check that the final version of the snapshot and targets roles is v2.
+This is a not so obvious use-case to keep in mind when thinking about updates. You can see how it looks in practice in the [reference implementation](https://github.com/theupdateframework/python-tuf/blob/develop/tests/test_updater_top_level_update.py#:~:text=test_expired_metadata).
+
+### Example rollback protection check with expired metadata:
+
+Now let’s see if a rollback attack protection can be performed when the local timestamp has expired. In this case we need at least two timestamp and snapshot versions, an expired older version of timestamp, and a verification that a rollback check is performed with the old version.
+
+For a timestamp rollback, the case is pretty similar to the use of expired metadata. We can do the following:
+1. Set timestamp v1 expiry one week ahead (Day 7)
+2. Perform refresh on the very first day
+3. Publish timestamp v2 with expiry three weeks ahead (Day 21)
+4. Perform a refresh somewhere between Day 7 and Day 21
+5. Verify that rollback check uses the expired timestamp v1. (For reference, see the implementation [example](https://github.com/theupdateframework/python-tuf/blob/develop/tests/test_updater_top_level_update.py#:~:text=test_expired_timestamp_version_rollback)).
+
+A similar approach can be used when testing both timestamp and snapshot rollback protection. We just need to guarantee that after the last snapshot update, the snapshot version is not the latest in order to verify a rollback check is performed both with expired timestamp and an older snapshot. Sounds complicated, but it’s pretty easy with the simulator and [this example](https://github.com/theupdateframework/python-tuf/blob/develop/tests/test_updater_top_level_update.py#:~:text=test_expired_timestamp_snapshot_rollback) illustrates it pretty well. 
+
```

Comment:
Are these two examples really an edge cases? Aren't they (vaguely) defined by the client workflow in the spec?

I do agree that they are tricky to implement in a client and tricky to test, and that the RepositorySimulator does a great job in achieving the latter.

Reply:
I probably described it as an edge case :grimacing: with the logic that 
* we want to ensure that we only use valid metadata
* expired metadata is not valid
* **except** when were validating new metadata with an older metadata (in this case specifically old metadata can be expired but must be otherwise valid)

this is defined in spec, that's true